scripts/assign-colors.py

```diff
@@ -0,0 +1,79 @@
+
+import argparse
+from sys import stderr, stdout
+from augur.io import read_metadata
+import csv
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        description="Assign colours based on ordering and metadata TSVs. Outputs a TSV on STDOUT.",
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter
+    )
+    parser.add_argument('--ordering', type=str, required=True, metavar="TSV",
+                        help="input ordering file. First column: category name, second column: value")
+    parser.add_argument('--color-schemes', type=str, required=True, metavar="TSV",
+                        help="input color schemes file. Each line is a list of colour hexes, in order. Each line should have a different number of entries.")
+    parser.add_argument('--metadata', type=str, required=True, metavar="TSV",
+                        help="restrict colors to only those found in metadata")
+    parser.add_argument('--duplications', type=str, required=False, nargs="+", metavar="categoryX=categoryY",
+                        help="Duplicate the colours generated for categoryX under the name categoryY")
+    return parser.parse_args()
+
+def read_ordering(fname):
+    # copied from <https://github.com/nextstrain/ncov/blob/80c06c15f0669c5990e0cc03628deeeedca7b720/scripts/assign-colors.py>
+    assignment = {}
+    with open(fname) as f:
+        for line in f.readlines():
+            array = line.lstrip().rstrip().split("\t")
```

Comment:
```suggestion
            array = line.strip().split("\t")
```

Reply:
Yes - better! I'm going to merge without changes however to keep it identical to where it was copied from and thus (maybe?) make [ENH: Add augur colors subcommand based on assign-colors.py scripts](https://github.com/nextstrain/augur/issues/1185) smoother when we get round to it.